components/airtable/actions/create-single-record/create-single-record.js

```diff
@@ -8,15 +8,41 @@ module.exports = {
   type: "action",
   props: {
     airtable,
-    baseId: {type: "$.airtable.baseId", appProp: 'airtable'},
-    tableId: { type: '$.airtable.tableId', baseIdProp: 'baseId' },
-    record: { propDefinition: [airtable, "record"] },
+    baseId: {
+      type: "$.airtable.baseId",
+      appProp: "airtable",
+    },
+    tableId: {
+      type: "$.airtable.tableId",
+      baseIdProp: "baseId",
+    },
+    record: {
+      propDefinition: [
+        airtable,
+        "record",
+      ],
+    },
+    typecast: {
+      type: "boolean",
+      description: "The Airtable API will perform best-effort automatic data conversion from string values if the typecast parameter is `True`. Automatic conversion is disabled by default to ensure data integrity, but it may be helpful for integrating with 3rd party data sources.",
+    },
   },
   async run() {
-    const Airtable = require('airtable');
-    const base = new Airtable({apiKey: this.airtable.$auth.api_key}).base(this.baseId);
-    return (await base(this.tableId).create([{
-      fields: this.record
-    }]))[0]
+    const Airtable = require("airtable");
+    const a = new Airtable({
+      apiKey: this.airtable.$auth.api_key,
+    });
+    const table = a.base(this.baseId)(this.tableId);
+    const args = [
+      [
+        {
+          fields: this.record,
+        },
+      ],
+      {
+        typecast: this.typecast,
+      },
+    ];
+    return (await table.create(...args))[0];
```

Comment:
Since `table.create` accepts only 2 arguments, I suggest removing the varargs here:

```suggestion
    const recordsData = [
      {
        fields: this.record,
      },
    ];
    const params = {
      typecast: this.typecast,
    };
    const [
      createdRecord,
    ] = await table.create(recordsData, params);
    return createdRecord;
```